keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,641 @@
+
+### API Changes
+
+N/A
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+All existing test jobs that use cgroup v2 should continue to pass without any flakiness.
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The respective kubelet subcomponents already have unit tests cases to handle cgroup v2.
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+N/A
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+1. Monitor existing cgroup v2 jobs, e.g.
+  - https://testgrid.k8s.io/sig-node-release-blocking#ci-crio-cgroupv2-node-e2e-conformance
+
+2. Ensure all features coverage by running all tests on cgroup v2 (while some may still run on cgroup v1 to test back compatibility)
+
```

Comment:
Do we have a default test configuration or guideline. If so, we should update that to use cgroupv2 for new features and their tests.

Reply:
> Do we have a default test configuration or guideline. If so, we should update that to use cgroupv2 for new features and their tests.

@BenTheElder WDYT?


xref: https://github.com/kubernetes/k8s.io/issues/5276

Reply:
> Do we have a default test configuration or guideline

That's a super broad topic ...
We have both node and cluster e2es and those operate across many tools and configurations.

Eliminating something or changing defaults in _all_ of CI tends to be a very large effort because of the scale and variety, but changing the defaults in these tools would get us most of the way there. We can't dictate this though, each project will need to agree because they also have to consider the impact for their non-kubernetes-CI users, excluding kube-up.sh and the local clusters that inherit from CI.

We should definitely be ensuring some v1 coverage until it's actually not supported.

https://github.com/kubernetes/k8s.io/issues/5276 is relevant for local-up-cluster.sh, KIND, `kinder`, (""unit tests"" that are really integration tests -- not recommended but they exist), integration tests (not really relevant to sig node).

For cloudy cluster and node e2e ... there's a lot to cover here including kube-up.sh, kOps, cluster-api, kubespray(?),  ...

Really we're talking about the defaults in the node bringup for those tools. For the local tools we will probably wind up moving to v2 only for the CI clusters and SIG K8s Infra / Testing will need bandwidth in https://github.com/kubernetes/k8s.io/issues/5276

Reply:
> We should definitely be ensuring some v1 coverage until it's actually not supported.

My comment is mostly for adding new tests for new features. Since the KEP aims to put cgroupv1 into the maintenance mode, it'd be good to ensure folks don't add necessary new tests using cgroupv1. Maybe we can start with node E2Es since the "new node features" should be tested there.

Reply:
@yujuhong thanks, I added a line here as well in `Graduation Criteria` to make cgroup v2 host requirement mandatory for new e2e and node e2e jobs. We can update the documentation in `test-infra` to reflect that.